packages/eventstream-serde-universal/src/getChunkedStream.ts

```diff
@@ -0,0 +1,107 @@
+export function getChunkedStream(
+  source: AsyncIterable<Uint8Array>
+): AsyncIterable<Uint8Array> {
+  let currentMessageTotalLength = 0;
+  let currentMessagePendingLength = 0;
+  let currentMessage: Uint8Array | null = null;
+  let messageLengthBuffer: Uint8Array | null = null;
+  const allocateMessage = function (size: number) {
+    if (typeof size !== "number") {
+      throw new Error(
+        "Attempted to allocate an event message where size was not a number: " +
+          size
+      );
+    }
+    currentMessageTotalLength = size;
+    currentMessagePendingLength = 4;
+    currentMessage = new Uint8Array(size);
+    const currentMessageView = new DataView(currentMessage.buffer);
+    currentMessageView.setUint32(0, size, false); //set big-endian Uint32 to 0~3 bytes
+  };
+
+  const iterator = async function* () {
+    const sourceIterator = source[Symbol.asyncIterator]();
+    let complete = false;
```

Comment:
Instead of variable `complete`, use `while(true)` and `break;` when `done` is false

Reply:
Resolved in https://github.com/aws/aws-sdk-js-v3/pull/1082/commits/8e9bbc011160d4bf0fd0e799cf4873a875d95c12